### PR TITLE
[PVKZU-11061] Fix no Jira ticket spamming

### PR DESCRIPTION
**Pull request by d-feller**

### CODE REVIEW

File: src/Console/Command/PurgeIssueFolderCommand.php [hunk 1]
```diff
@@ -136,25 +142,19 @@ protected function findBranchesToCleanup(
...
-                    }
-                } elseif ($statusFound) {
-                    $remove[] = $dir;
-                }
-            } else {
+            $issue = null;
+            try {
+                $issue = $this->fetchIssueUseCase->execute($dir->getFilename(), $branchNameFilter);
+            } catch (JiraTicketNotFoundException $e) {
                 $notfound[] = $dir;
```
Comment (icanhazstring) [inline src/Console/Command/PurgeIssueFolderCommand.php:151]:
```suggestion
                $remove[] = $dir;
```

I think the folder should be removed if the jira ticket was removed or never existed.
Or else we would end up with dead folders that would be scanned all over again.

File: src/Exception/JiraTicketNotFoundException.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace duckpony\Exception;
+
+class JiraTicketNotFoundException extends \Exception
```
Comment (icanhazstring) [inline src/Exception/JiraTicketNotFoundException.php]:
```suggestion
class JiraTicketNotFoundException extends \RuntimeException
```

Please always use a more specific exception as base for custom ones.

File: src/UseCase/FetchIssueUseCase.php [hunk 3]
```diff
@@ -32,11 +36,17 @@ public function execute(string $issueName, array $issueNameFilter = []): ?Issue
             /** @var Issue $issue */
             $issue = $this->issueService->get($issueName, ['fields' => ['status']]);
         } catch (JiraException $e) {
+            if ($e->getCode() === 404) {
+                throw new JiraTicketNotFoundException(
+                    'Jira ticket does not exist or was deleted',
+                    $e->getCode()
+                );
```
Comment (icanhazstring) [inline src/UseCase/FetchIssueUseCase.php]:
```suggestion
                throw new JiraTicketNotFoundException(
                    'Jira ticket does not exist or was deleted',
                    $e->getCode(),
                    $e
                );
```

As a small rule of thumb. If you catch a system foreign exception and want to rethrow it with you system internal one. Always add the previous exception to the stacktrace or you will hide the source exception which makes it harder to track down problems.

File: test/Unit/Rule/IsIssueWithGivenStatusRuleTest.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
+<?php
```
Comment (icanhazstring) [inline test/Unit/Rule/IsIssueWithGivenStatusRuleTest.php:1]: Nice one. Thanks 👍

**Review by icanhazstring — CHANGES_REQUESTED** [review]
Thanks! Small changes needed tho.

File: src/Console/Command/PurgeIssueFolderCommand.php (hunk 1 above)
Comment (d-feller) [reply src/Console/Command/PurgeIssueFolderCommand.php:151]: The folders in $notfound are being deleted in a later step in Line 120: `$this->removeNotFound($notfound, $io, $yes);`

Comment (icanhazstring) [reply src/Console/Command/PurgeIssueFolderCommand.php:151]: Ok, than its fine and we can leave it that way :)

**Pull request merged**